sdk/state/hash.go

```diff
@@ -0,0 +1,35 @@
+package state
+
+import (
+	"encoding/hex"
+	"fmt"
+)
+
+type TransactionHash [32]byte
+
+func (h TransactionHash) String() string {
```

Comment:
❓  for my own understanding where/how are these methods called? I'm assuming somewhere implicitly

Reply:
The `String` function is an implementation of the fmt.Stringer interface, so if this hash is formatted into a string, fmt.Print* functions can print a hex version rather than a byte array. It's just for improved debugging. https://pkg.go.dev/fmt#Stringer